jmix-reports/reports/src/main/java/io/jmix/reports/libintegration/JmixObjectToStringConverter.java

```diff
@@ -66,12 +68,28 @@ public Object convertFromString(Class parameterClass, @Nullable String paramValu
         } else if (String.class.isAssignableFrom(parameterClass)) {
             return paramValueStr;
         } else if (Entity.class.isAssignableFrom(parameterClass)) {
-            UUID id = UUID.fromString(paramValueStr);
-            Optional entityOpt = dataManager.load(parameterClass)
-                    .id(id)
-                    .fetchPlan(FetchPlan.BASE)
-                    .optional();
-            return entityOpt.isPresent() ? entityOpt.get() : null;
+            MetaProperty idProperty = metadataTools.getPrimaryKeyProperty(parameterClass);
+            Object idValue = null;
+
+            if (idProperty != null && idProperty.getRange().isDatatype()) {
```

Comment:
Please check what happens if primary key is not a Datatype, e.g. the entity has a composite PK. It seems that the method will just return null. We should either handle composite primary keys or raise an exception saying that primary key type is not supported.

Reply:
Now an exception will be thrown.